jupyter_kernel_mgmt/subproc/manager.py

```diff
@@ -42,44 +43,44 @@ def __init__(self, popen, files_to_cleanup=None, win_interrupt_evt=None):
         self.log = get_app_logger()
         self.kernel_id = str(uuid.uuid4())
 
-    def wait(self, timeout):
+    async def wait(self, timeout):
         """"""
         if timeout is None:
             # Wait indefinitely
-            self.kernel.wait()
+            await maybe_future(self.kernel.wait())
             return False
 
         if six.PY3:
             try:
-                self.kernel.wait(timeout)
+                await maybe_future(self.kernel.wait(timeout))
                 return False
             except subprocess.TimeoutExpired:
                 return True
         else:
             pollinterval = 0.1
             for i in range(int(timeout / pollinterval)):
                 if self.is_alive():
-                    time.sleep(pollinterval)
+                    await asyncio.sleep(pollinterval)
                 else:
                     return False
-            return self.is_alive()
+            return await self.is_alive()
 
-    def cleanup(self):
+    async def cleanup(self):
         """Clean up resources when the kernel is shut down"""
         # cleanup connection files on full shutdown of kernel we started
         for f in self.files_to_cleanup:
             try:
-                os.remove(f)
+                await maybe_future(os.remove(f))
```

Comment:
I don't think `os.remove()` can ever be async?

Reply:
Probably correct - updated in next rev.